tlsutil/config.go

```diff
@@ -199,13 +199,15 @@ type Configurator struct {
 	https       protocolConfig
 	internalRPC protocolConfig
 
-	// autoTLS stores configuration that is received from the auto-encrypt or
-	// auto-config features.
+	// autoTLS stores configuration that is received from:
+	// - The auto-encrypt or auto-config features for client agents
+	// - The servercert.CertManager for server agents.
 	autoTLS struct {
 		extraCAPems          []string
 		connectCAPems        []string
 		cert                 *tls.Certificate
 		verifyServerHostname bool
+		peeringServerName    string
```

Comment:
nit, please feel free to ignore:

What do you think of calling this `serverName`? (and the method below `UpdateAutoTLSServerName`).

From the TLS configurator's point of view, there's nothing peering-specific about this. It's just that sometimes the automatically provisioned certificate has an associated server name, which (when it matches the client's request) causes it to be preferred over the manually provisioned one.

Reply:
I'm OK with switching it to `serverName`.

My thinking was that there may be additional server names to handle this way in the future (maybe for agentless?)

Reply:
Oh, maybe! For agentless, right now we're imagining folks will use "real" TLS certs, signed by a public CA, so they don't need to configure roots on the "client".